### PR TITLE
Mercenary trading

### PR DESCRIPTION
**Pull request by Steampunkery**
# Description
This PR overhauls pretty much the entire mercenary system, with exceptions noted below the major changes. 

## Major Changes

1. Mercenaries are now created as `SolShip`s instead of `FarShip`s. This change counts as major because it opens mercenaries up to a wider range of possibilities. With this change, mercenaries can drop loot and have items equipped easily, just like the player.
2. Four new mercenary interaction screens were added. You'll notice if you run the game with this PR, under the usual buttons on the left side just below the "Talk" button is a button labeled "Mercs". This button has a hotkey binding of "M" and will flash like the "Items" button when there is a new mercenary added. Mercenaries will also be removed from this screen when they die. Below is a rundown of the new screens:

    - ChooseMercenary.java - This is the screen that is called up when the "Mercs" button is pressed that allows the player to select a mercenary and an operation to perform on that mercenary.
    - GiveItems.java - This screen is used when the player selects the "Give Items" operation. It shows the player's inventory and allows him/her to give items to the selected mercenary
    - TakeItems.java - This screen is opened when the player chooses the "Take Items" operation. It shows the mercenary's inventory and allows the player to take items from the merc.
    - EquipItems.java - Probably the most important screen because the other two would be useless without it. It opens at the click of "Equip Items" and shows the mercenary's inventory. It facilitates equipping items to the mercenary, like guns and shields.

## Exceptions
There is one exceptions to this overhaul as I mentioned above. The mercenary save system has barely been touched. I will overhaul that system as another PR because of the amount of changes. As it currently stands, mercenaries and items you gave them simply disappear when you exit the game.

# Testing
All of the features are easily testable. The best way that I found is to spawn as a Formic Capital ship because it has enough money to buy three mercenaries from it's starting station. You will be able to see each new mercenary appear in the [M]ercs screen as you buy them. Each mercenary can be traded with individually. To test removal of a dead mercenary, simply use the massive bulk of the Formic Capital to smash a merc against the space station. 

# Notes
Formic hire prices were changed because they were intentionally left out for mercenary testing purposes, but when this gets merged, they'll need real hire prices.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by vampcat** [comment]
@GooeyHub: ok to test

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by vampcat** [comment]
@Steampunkery Can you have a look at the CodeFactor issues, and see if they're relevant/fixable?

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/game/item/MercItem.java [hunk 1]
```diff
@@ -50,8 +54,12 @@ public SolItem copy() {
     }
 
     @Override
+    /**
```
Comment (vampcat) [inline engine/src/main/java/org/destinationsol/game/item/MercItem.java]: Do you mind shifting this javadoc to the parent class?

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**